[DOCS-1091] Docs for toggling between automatic and manual workspaces

mdlinville opened the pull request:
## Description

[DOCS-1091] Docs for toggling between automatic and manual workspaces

- Clearing all panels is now possible only in a manual workspace
- Document resetting automatic workspaces
- Document duplicating a panel (missing)
- Update docs about checking workspace settings, remove outdated graphics
- Some restructuring
## Ticket

Does this PR fix an existing issue? If yes, provide a link to the ticket here:

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [x] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [x] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [x] I merged the latest changes from `main` into my feature branch before submitting this PR.


[DOCS-1091]: https://wandb.atlassian.net/browse/DOCS-1091?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mdlinville committed:
[DOCS-1091] Docs for toggling between automatic and manual workspaces

- Clearing all panels is now possible only in a manual workspace
- Document resetting automatic workspaces
- Document duplicating a panel (missing)
- Update docs about checking workspace settings, remove outdated graphics

mdlinville committed:
Add workspace's to Vale vocab

mdlinville committed:
Style edits

mdlinville commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -9,6 +9,33 @@ import TabItem from '@theme/TabItem';
+
+A workspace's type indicates whether the workspace generates panels automatically. Next to the name of your workspace is a clipboard icon.
+
+- If the icon is a green diamond, automatic panel generation is enabled.
```
@johndmulhausen Can we add exceptions for these? I can't think of a way for it not to be passive!

johndmulhausen replied:
How about: A green diamond icon indicates that the system is configured for automatic panel generation.

mdlinville commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -9,6 +9,33 @@
+A workspace's type indicates whether the workspace generates panels automatically. Next to the name of your workspace is a clipboard icon.
+
+- If the icon is a green diamond, automatic panel generation is enabled.
+- If the icon is a purple safety pin and ruler, automatic panel generation is disabled.
```
@johndmulhausen Ditto

johndmulhausen replied:
A purple safety pin and ruler icon indicates that automatic panel generation is turned off.

mdlinville committed:
Add disable and disabled to Vale vocab

mdlinville committed:
Fix typo

mdlinville committed:
Reword to avoid passive voice

mdlinville commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -9,6 +9,33 @@
+A workspace's type indicates whether the workspace generates panels automatically. Next to the name of your workspace is a clipboard icon.
+
+- If the icon is a green diamond, automatic panel generation is enabled.
+- If the icon is a purple safety pin and ruler, automatic panel generation is disabled.
```
I don't know why https://github.com/wandb/docodile/pull/941/commits/2ddaf2ebcddf3880c8a028bab8f801543f3633c9 did not resolve this one

johndmulhausen replied:
I believe you have to manually hit "resolve?"

johndmulhausen replied:
Actually we do not want to accept "disabled" as a term, e.g. https://developers.google.com/style/inclusive-documentation#ableist-language

mdlinville replied:
See https://github.com/wandb/docodile/pull/941#discussion_r1872052545. I'm taking your suggestion here, but I can imagine some contexts in which the prose becomes tortured to work around this.

johndmulhausen commented on .github/styles/config/vocabularies/Docs/accept.txt:
```diff
@@ -20,6 +20,8 @@ CTAButtons
 [Dd]equeues
 [Dd]eserialize
+[Dd]isable
+[Dd]isabled
```
```suggestion
```
https://developers.google.com/style/inclusive-documentation#ableist-language

mdlinville replied:
And yet: https://cloud.google.com/service-usage/docs/enable-disable

It is hard for me to imagine being able to avoid the words "enable" and "disable" across an entire docs corpus. WDYT?

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1091

mdlinville committed:
John's style guide feedback

mdlinville committed:
Update .github/styles/config/vocabularies/Docs/accept.txt

Co-authored-by: John Mulhausen <john.mulhausen@wandb.com>

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1091' into DOCS-1091

mdlinville committed:
More active voice

mdlinville committed:
Typo

mdlinville commented:
Pausing to consider feedback about the broader structure of the content about panels. https://www.notion.so/wandbai/Panel-management-docs-153e2f5c7ef380739e6ee10237af94ea

mdlinville committed:
Updates and reorg:

- Add section for managing panels
- Reorganize section about adding panels
- Add details about moving panels in automatic vs manual workspaces

mdlinville committed:
Adjust content about
- adding multiple panels
- changing a panel's type
- Removing all panels

mdlinville committed:
Style stuff

mdlinville commented:
Ready for further review, @Cecile0112358 @sephmard

mdlinville committed:
Remove Check Workspace Mode section, adjust workspace type to workspace mode

mdlinville committed:
Adjust intro to add more details about when to choose each workspace mode

mdlinville committed:
Adjust docs about workspace setting defaults

mdlinville committed:
Address most feedback from https://www.notion.so/wandbai/Panel-management-docs-153e2f5c7ef380739e6ee10237af94ea#153e2f5c7ef3809cb4d1e6128d925e85

mdlinville commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -7,101 +7,135 @@ import Tabs from '@theme/Tabs';
 
-<!-- ## Add a section -->
+1. To expand or collapse all sections, click the `...` menu next to the panel search field, then select **Expand all sections** or **Collapse all sections**.
+1. To add a section, click **Add section**. To add a new section above or below an existing section, you can instead click the section's `...` menu, then click **New section below** or **New section above**.
```
I'll ignore this warning for now

mdlinville commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -7,101 +7,135 @@
 
-<!-- ## Add a section -->
+1. To expand or collapse all sections, click the `...` menu next to the panel search field, then select **Expand all sections** or **Collapse all sections**.
+1. To add a section, click **Add section**. To add a new section above or below an existing section, you can instead click the section's `...` menu, then click **New section below** or **New section above**.
```
I'll ignore this warning for now because it's literally what the UI says

mdlinville committed:
Updates based on Notion

mdlinville committed:
Style nit

mdlinville commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -7,101 +7,113 @@ import Tabs from '@theme/Tabs';
+
+To add a section, click **Add section** after the last section.
+
+To add a new section before or after an existing section, you can instead click the section's action `...` menu, then click **New section below** or **New section above**.
```
Ignoring because that's what the UI says.

mdlinville committed:
More style stuff

mdlinville committed:
Increase icon heights

mdlinville committed:
Update workspace mode descriptions to a table

Cecile0112358 commented:
Looks good to go!

ngrayluna commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -7,101 +7,114 @@ import Tabs from '@theme/Tabs';
+| Icon | Workspace mode |
+| --- | --- |
+| <img src="/images/app_ui/automated_workspace.svg" alt="automated workspace icon" width="32px" /> | **Automated workspaces** automatically generate panels for all keys logged in the project. This can help you get started by visualizing all available data for the project. |
+| <img src="/images/app_ui/manual_workspace.svg" alt="manual workspace icon" width="32px" /> | **Manual workspaces** start as blank slates and display only those panels intentionally added by users. Choose a manual workspace when you  care mainly about a fraction of the keys logged in the project, or for a more focused analysis. |
```
nit: extra space between "you" and "care"

ngrayluna commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -7,101 +7,114 @@ import Tabs from '@theme/Tabs';
+
+1. Hover your mouse over the panel.
+2. Select the action `...` menu.
+3. Click **Delete**
```
nit: missing a sentence period.

ngrayluna commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -7,101 +7,114 @@ import Tabs from '@theme/Tabs';
-3. From the dropdown, select a type of panel to add
-![](/images/app_ui/add_single_panel.gif) 
-4. (Optional) If prompted, define parameters for the plot. 
+Weights and Biases projects support two different workspace modes. The icon next to the workspace name shows its mode. 
```
So this is a weird one... TL;DR branding told us to not use the long form of "Weights & Biases" and instead use "W&B". This may change in the near future.

ngrayluna commented on docs/guides/app/features/panels/intro.md:
```diff
@@ -7,101 +7,114 @@ import Tabs from '@theme/Tabs';
-  <TabItem value="quick">
+| Icon | Workspace mode |
+| --- | --- |
+| <img src="/images/app_ui/automated_workspace.svg" alt="automated workspace icon" width="32px" /> | **Automated workspaces** automatically generate panels for all keys logged in the project. This can help you get started by visualizing all available data for the project. |
```
In the future, might be helpful to define (a quick one-liner) what we mean by "keys".  Depending on who the target audience is, they might confuse key-item pairs passed to  `wandb.init.log`  with SSH keys, etc..

ngrayluna replied:
I dig the icons within tables.

ngrayluna reviewed (approved):
Minor nits.

mdlinville committed:
Most of Noah's feedback, re-number lists

mdlinville committed:
Add link to define logged data
